src/classes/CountryStateDistrictDataManipulation.cls

```diff
@@ -0,0 +1,209 @@
+/**
+*  Description     :  This Class Used to fill Map Data Dynamically from Country, State and Districts. 
+*                     Aggregate Functions used to query and update total number of district and states on Country Object.
+*
+*  Created By      :  Mrinal Das
+*
+*  Created Date    :  07/13/2022
+*
+*  Revision Logs   :    V1.0 - Created - Mrinal Das
+*				    
+*
+**/
+
+public with sharing class CountryStateDistrictDataManipulation {
+    
+    // Global Map
+    Map<String,Map<String,List<String>>> mapOfCountryWithStateAndDistrict = new Map<String,Map<String,List<String>>>();
+
+    //Method to Map Districts, States and Countries
+    public CountryStateDistrictDataManipulation(String word) {
+        
+            List<District__c> listOfdistrict = [SELECT Name, State_Name__r.Name, State_Name__r.Country_Name__r.Name FROM District__c];
```

Comment:
map filling concept  + AggregateResult Concept will be in Constructor and finding the country, state, and District logic will be in the Parameterized Constructor.